hal/halx86/xbox/part_xbox.c

```diff
@@ -118,20 +119,38 @@ HalpXboxDeviceHasXboxPartitioning(IN PDEVICE_OBJECT DeviceObject,
         return STATUS_NO_MEMORY;
     }
 
+    Offset.QuadPart = 0;
+    Status = HalpXboxReadSector(DeviceObject, SectorSize, &Offset, SectorData);
+    if (!NT_SUCCESS(Status))
+    {
+        goto Cleanup;
+    }
+
+    HasMBRPartitioning = (*((UCHAR *) SectorData + SectorSize - 2) == 0x55 && *((UCHAR *) SectorData + SectorSize - 1) == 0xAA);
```

Comment:
To be able to find quickly all the instances of 0x55aa usage in the code, maybe read this SectorData + SectorSize into a USHORT variable, that you then compare with 55aa (careful about the order of the bytes).

Reply:
Improved (used existing definition `PARTITION_SIGNATURE`).